secboot/secboot_tpm.go

```diff
@@ -481,6 +483,86 @@ func ProvisionForCVM(initramfsUbuntuSeedDir string) error {
 	return nil
 }
 
+func withTPMConnection(fn func(tpm *sb_tpm2.Connection)) error {
+	tpm, err := sbConnectToDefaultTPM()
+	if err != nil {
+		return fmt.Errorf("cannot connect to TPM: %v", err)
+	}
+	defer tpm.Close()
+	if !isTPMEnabled(tpm) {
+		return fmt.Errorf("TPM device is not enabled")
+	}
+
+	// Workaround for secboot to reuse opened tpm connection.
+	old := sb_tpm2.ConnectToTPM
+	sb_tpm2.ConnectToTPM = func() (*sb_tpm2.Connection, error) {
+		return tpm, nil
+	}
+	defer func() { sb_tpm2.ConnectToTPM = old }()
+
+	fn(tpm)
+	return nil
+}
+
+func kdfOptions(volumesAuth device.VolumesAuthOptions) (sb.KDFOptions, error) {
```

Comment:
this probably should take a pointer as well to device.VolumesAuthOptions)

Reply:
updated, thanks!